pkg/sql/catalog/lease/lease.go

```diff
@@ -122,10 +123,24 @@ func (m *Manager) WaitForOneVersion(
 	ctx context.Context, id descpb.ID, retryOpts retry.Options,
 ) (desc catalog.Descriptor, _ error) {
 	for lastCount, r := 0, retry.Start(retryOpts); r.Next(); {
-		if err := m.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) (err error) {
+		if err := m.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 			version := m.storage.settings.Version.ActiveVersion(ctx)
-			desc, err = catkv.MustGetDescriptorByID(ctx, version, m.Codec(), txn, nil /* vd */, id, catalog.Any)
-			return err
+			// Use the lower-level GetCrossReferencedDescriptorsForValidation to avoid
+			// performing cross-descriptor validation while waiting for leases to
+			// drain. On the one hand, it's somewhat expensive. More importantly,
+			// there are valid cases where descriptors can be removed or made invalid,
+			// and we don't particularly care about them when using this method.
```

Comment:
nit: can we give a more specific example in the comment of "valid cases where descriptors can be removed or made invalid" so that the reader knows better why we don't want to do cross-descriptor validation when waiting for one lease.

Reply:
Added an example.